app/presenters/checkbox_question_presenter.rb

```diff
@@ -34,4 +34,16 @@ def checkboxes
       }
     end
   end
+
+  def checked?(value)
+    return if response_for_current_question.blank?
+
+    response = response_for_current_question
+
+    if response_for_current_question.is_a?(String)
+      response = to_response(response_for_current_question)
+    end
```

Comment:
This string check feels pretty low level, do we know what circumstances it comes up as a concern? The past code looks like it is only needed on `params[:previous_response]` so I'm wondering if this can be dealt with within: https://github.com/alphagov/smart-answers/blob/d2f1da06b6b3c44ba97144baedec3db0757bd9de/app/presenters/flow_presenter.rb#L86 ?

Reply:
Is the suggestion to call `to_response` on `params[:previous_response]` in the `FlowPresenter` regardless of the question type?

Reply:
That could work, because for most cases it does nothing, and for the others, I'm having to call it anyway.

Reply:
Although, `to_response` only exists in the QuestionPresenters, not in FlowPresenter, so I don't think I can easily use it.